File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2024-05-01/examples-java/ApiManagementCreateWorkspacePolicyFragment.java

import com.azure.resourcemanager.apimanagement.fluent.models.PolicyFragmentContractInner;
import com.azure.resourcemanager.apimanagement.models.PolicyFragmentContentFormat;

/**
 * Samples for WorkspacePolicyFragment CreateOrUpdate.
 */
public final class Main {
    /*
     * x-ms-original-file:
     * specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2024-05-01/examples/
     * ApiManagementCreateWorkspacePolicyFragment.json
     */
    /**
     * Sample code: ApiManagementCreateWorkspacePolicyFragment.
     * 
     * @param manager Entry point to ApiManagementManager.
     */
    public static void apiManagementCreateWorkspacePolicyFragment(
        com.azure.resourcemanager.apimanagement.ApiManagementManager manager) {
        manager.workspacePolicyFragments().createOrUpdate("rg1", "apimService1", "wks1", "policyFragment1",
            new PolicyFragmentContractInner()
                .withValue("<fragment><json-to-xml apply=\"always\" consider-accept-header=\"false\" /></fragment>")
                .withDescription("A policy fragment example").withFormat(PolicyFragmentContentFormat.XML),
            null, com.azure.core.util.Context.NONE);
    }
}

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2024-05-01/examples-js/ApiManagementCreateWorkspacePolicyFragment.js
const { ApiManagementClient } = require("@azure/arm-apimanagement");
const { DefaultAzureCredential } = require("@azure/identity");
require("dotenv/config");

/**
 * This sample demonstrates how to Creates or updates a policy fragment.
 *
 * @summary Creates or updates a policy fragment.
 * x-ms-original-file: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2024-05-01/examples/ApiManagementCreateWorkspacePolicyFragment.json
 */
async function apiManagementCreateWorkspacePolicyFragment() {
  const subscriptionId =
    process.env["APIMANAGEMENT_SUBSCRIPTION_ID"] || "00000000-0000-0000-0000-000000000000";
  const resourceGroupName = process.env["APIMANAGEMENT_RESOURCE_GROUP"] || "rg1";
  const serviceName = "apimService1";
  const workspaceId = "wks1";
  const id = "policyFragment1";
  const parameters = {
    format: "xml",
    description: "A policy fragment example",
    value: '<fragment><json-to-xml apply="always" consider-accept-header="false" /></fragment>',
  };
  const credential = new DefaultAzureCredential();
  const client = new ApiManagementClient(credential, subscriptionId);
  const result = await client.workspacePolicyFragment.beginCreateOrUpdateAndWait(
    resourceGroupName,
    serviceName,
    workspaceId,
    id,
    parameters,
  );
  console.log(result);
}
